### PR TITLE
Move delta in `preshard`, `LiftGlobalBufferAlloc` and `HuggingFaceLoader` to ollm `slm/` and clean-up 

### PR DESCRIPTION
**Pull request by sunggg**
This PR moves stable files to https://github.com/octoml/ollm/tree/main/mlc-serve/slm so that we can centralize the delta from upstream in one place. 

FP8 flow and SmoothQuant will be moved as well once they are stabilized. cc. @csullivan @elvin-n 

@Lunderberg, for the fix on `LiftGlobalBufferAlloc`, please send it to either upstream mlc-ai or ollm.

### CODE REVIEW

**Comment by elvin-n** [comment]
the file `python/mlc_llm/quantization/smoothquant_utils.py` should be migrated to ollm repo as well since it it not required for smq in mlc_llm
cc @ibsidorenko

File: python/mlc_llm/loader/huggingface_loader.py [hunk 1]
```diff
@@ -99,10 +98,7 @@ def __init__(
         check_parameter_usage(extern_param_map, set(self.torch_to_path.keys()))
 
     def load(
-        self,
-        device: Device,
-        preshard_funcs: Dict[str, Callable] = None,
-        smoothquant_funcs: Dict[str, Callable] = None,
+        self, device: Device, preshard_funcs: Dict[str, Callable] = None
```
Comment (ibsidorenko) [inline python/mlc_llm/loader/huggingface_loader.py:101]: hmm... SmoothQuant was removed. Is it intentional?
`smoothquant_funcs: Dict[str, Callable] = None,`

Comment (sunggg) [reply python/mlc_llm/loader/huggingface_loader.py:101]: This is paired with ollm PR: https://github.com/octoml/ollm/pull/756
You can find it in the `slm/loader/huggingface_loader.py`.

**Comment by sunggg** [comment]
@elvin-n, would you mind moving that with all the other smq files to put everything related to smq in the single PR? I think it would be easier to track in the future.

**Comment by sunggg** [comment]
Since there is no objection, I'm going to merge this. (This is strictly matter of changing file structure, no functional change at all)
As of today, the diff with the last merge point of upstream (`d3d264d4b05d73e9757375013b842254f052c6ed`) is as follows:
```
(mlc-serve) spark@ip-172-31-7-51:~/ollm/deps/mlc-llm$ git diff --name-only HEAD d3d264d4b05d73e9757375013b842254f052c6ed 
3rdparty/tvm
python/mlc_llm/__init__.py
python/mlc_llm/compiler_pass/estimate_memory_usage.py
python/mlc_llm/interface/compiler_flags.py
python/mlc_llm/interface/convert_weight.py
python/mlc_llm/model/llama/llama_quantization.py
python/mlc_llm/model/mistral/mistral_quantization.py
python/mlc_llm/model/mixtral/mixtral_quantization.py
python/mlc_llm/model/model.py
python/mlc_llm/op/moe_matmul.py
python/mlc_llm/quantization/__init__.py
python/mlc_llm/quantization/awq_quantization.py
python/mlc_llm/quantization/fp8_quantization.py
python/mlc_llm/quantization/ft_quantization.py
python/mlc_llm/quantization/group_quantization.py
python/mlc_llm/quantization/per_tensor_quantization.py
python/mlc_llm/quantization/quantization.py
python/mlc_llm/quantization/smooth_quantization.py
python/mlc_llm/quantization/smoothquant_utils.py
python/mlc_llm/quantization/utils.py
python/mlc_llm/support/tensor_parallel.py
setup.py
```
Others will follow-up with the FP8 PTQ and SMQ flows.

**Pull request merged**